Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -592,9 +590,7 @@ void SourceBuffer::sourceBufferPrivateAppendComplete(AppendResult result)
     m_source->monitorSourceBuffers();
     m_private->reenqueueMediaIfNeeded(m_source->currentTime());
 
-    StringPrintStream message;
-    message.printf("SourceBuffer::sourceBufferPrivateAppendComplete(%p) - buffered = %s", this, toString(m_private->buffered()->ranges()).utf8().data());
-    DEBUG_LOG(LOGIDENTIFIER, message.toString());
+    DEBUG_LOG(LOGIDENTIFIER, "buffered = ", m_private->buffered()->ranges().toString());
```

Comment:
This is fine as-is, but I _think_ it should be possible to drop the final `.toString()`, with the addition of a LogArgument template for PlatformTimeRanges.

Reply:
It is, yes, but I was doing it wrong.

I tried, the build failed with an indecipherable template error and I just gave up and used `toString()`. After digging a bit more, it turns out that I [already had the `LogArgument<PlatformTimeRanges>::toString()` code in place](https://github.com/WebKit/WebKit/pull/6044/commits/e1ccf51b4a4700f889d6529041a27a5b14da9bfa#diff-0766b78fb2791875107e1415c7aeae8098e7dbca2862b9f4b68c7d6bb5ab794bR170), but defined in the WebCore namespace instead of the WTF one.

I've modified the definition in `PlatformTimeRanges.h` to be on WTF and now it works (see new patch). I was wondering why [it worked out of the box for MediaTime](https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/MediaTime.h#L193) and not for PlatformTimeRanges until I realized that MediaTime belongs to WTF.